pkg/envoy/bootstrap/envoy_config_health_probes.go

```diff
@@ -89,34 +89,32 @@ func getProbeCluster(clusterName string, port int32) *xds_cluster.Cluster {
 	}
 }
 
-func getLivenessListener(originalProbe *models.HealthProbe) (*xds_listener.Listener, error) {
-	if originalProbe == nil {
-		return nil, nil
-	}
-	return getProbeListener(livenessListener, livenessCluster, constants.LivenessProbePath, constants.LivenessProbePort, originalProbe)
+type probeListenerRoute struct {
+	pathPrefixMatch   string
+	clusterName       string
+	pathPrefixRewrite string
+	timeout           time.Duration
 }
 
-func getReadinessListener(originalProbe *models.HealthProbe) (*xds_listener.Listener, error) {
-	if originalProbe == nil {
-		return nil, nil
-	}
-	return getProbeListener(readinessListener, readinessCluster, constants.ReadinessProbePath, constants.ReadinessProbePort, originalProbe)
+type probeListenerBuilder struct {
```

Comment:
Having both a models.HealthProbe, and a `probeListenerBuilder` seems like an unnecessary layer of indirection. Can we reduce to one or the other?

Reply:
`models.HealthProbe` represents a set of 3 probes: startup, listener, and readiness. The `probeListenerBuilder` builds an Envoy listener for any generic probe based on the information set on the builder. I don't see the redundancy here; could you expound your thought process?

Reply:
It looks like `getProbeListener` was replaced with `probeListenerBuilder`.   `getProbeListener` could be reused reducing the indirection and remove some of the additional mapping above in `getProbeResources()`.  

I think the pro for that approach would be a little less mapping code in `getProbeResources`?

Reply:
I'm not sure how `getProbeListener` could be reused here seeing as all that logic moved to `Build`?

Reply:
moved to build in the `probeListenerBuilder`?  I think the suggestion was to drop that but might be miss-understanding.    I could go either way personally, what is here is pretty readable but I see how it adds some additional mapping